### PR TITLE
KB/feat/add stack

### PR DESCRIPTION
**Pull request by kryzbone**

### CODE REVIEW

File: Alembic/versions/3df848fa09a8_create_stacks.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+down_revision = '99f4fc5a9a7b'
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('stacks',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('name', sa.String(), nullable=True),
```
Comment (TMCreme) [inline Alembic/versions/3df848fa09a8_create_stacks.py:23]: Shouldn't this be a required field? I think making the name field nullable does not make sense. Kindly confirm why this is.

File: api/api_models/stacks.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from pydantic import BaseModel, Field
+from datetime import datetime
+
+
+class Stacks(BaseModel):
+	id: str = Field(...)
+	name: str = Field(...)
```
Comment (TMCreme) [inline api/api_models/stacks.py:7]: I think the name field should have a unique validation to prevent duplicates

File: api/api_models/stacks.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from pydantic import BaseModel, Field
+from datetime import datetime
+
+
+class Stacks(BaseModel):
+	id: str = Field(...)
```
Comment (TMCreme) [inline api/api_models/stacks.py:6]: Is it also possible to make this the primary key?

File: api/routes/stacks.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	db.add(new_stack)
+	db.commit()
+	db.refresh(new_stack)
+
+	return new_stack
+
+# read stack with id
+@stack_router.get("/{stack_id}", response_model=stack_schemas.Stacks)
+async def read_stack(stack_id: int, db: Session = Depends(get_db)):
+	stack = db.query(Stack).filter(Stack.id == stack_id).first()
```
Comment (TMCreme) [inline api/routes/stacks.py:39]: If we have a primary key on the table, we can use the `session.get(Stack, {stack_id})`

File: db/models/users.py [hunk 5]
```diff
@@ -94,3 +96,15 @@ class Feed(Base):
         Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
     user = relationship("User")
     
+
+class Stack(Base):
+    __tablename__ = 'stacks'
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(String)
```
Comment (TMCreme) [inline db/models/users.py:103]: I think this should have a unique validation and nullable=False as well

File: test/test_stacks.py
Comment (TMCreme) [inline test/test_stacks.py:1]: I think we should test for duplicate record (in the sense of what field (probably the name field) constitutes duplicates)

**Review by TMCreme — COMMENTED** [review]
@kryzbone Kindly check my comments.

File: Alembic/versions/3df848fa09a8_create_stacks.py (hunk 1 above)
Comment (kryzbone) [reply Alembic/versions/3df848fa09a8_create_stacks.py:23]: will fix this. taught the default was false

File: api/routes/stacks.py (hunk 4 above)
Comment (kryzbone) [reply api/routes/stacks.py:39]: this is what is being used in the docs, havent found any use of a .get so not sure if that is possible 
https://fastapi.tiangolo.com/tutorial/sql-databases/#read-data

Comment (kryzbone) [reply api/routes/stacks.py:39]: this is what is being used in the docs, haven't found any use of a .get so not sure if that is possible 
https://fastapi.tiangolo.com/tutorial/sql-databases/#read-data

Comment (TMCreme) [reply api/routes/stacks.py:39]: Alright thanks.

File: test/test_stacks.py
Comment (kryzbone) [reply test/test_stacks.py:1]: added test for duplicates

**Review by TMCreme — APPROVED** [review]

**Pull request merged**